frontend/src/lib/components/SaveToDashboard/SaveToDashboardModal.tsx

```diff
@@ -27,7 +30,7 @@ export function SaveToDashboardModal({ visible, closeModal, insight }: SaveToDas
 
     async function save(event: MouseEvent | FormEvent): Promise<void> {
         event.preventDefault()
-        updateInsight({ ...insight, dashboard: dashboardId }, () => {
+        updateInsight({ ...insight, dashboards: [...(insight.dashboards ?? []), dashboardId] }, () => {
```

Comment:
Nit: I think this will not be exposed to anyone, but we could just do `dashboards: [dashboardId]` to make sure there's always just one per insight. We'll change this soon anyway, so 🤷